model/src/pyrenew/metaclass.py

```diff
@@ -94,6 +94,46 @@ def _assert_sample_and_rtype(
     return None
 
 
+class SampledValue:
```

Comment:
I thought the plan was to have `SampledValue`s be `NamedTuple`s to ensure that they are immutable.